### PR TITLE
upgrade django 1.8

### PR DESCRIPTION
**Pull request by muhammad-ammar**
TNL-3146

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,7 +12,6 @@ endif
 test: clean
 	./manage.py test --settings=$(test_settings) --with-coverage --with-ignore-docstrings \
 		--exclude-dir=notesserver/settings --cover-inclusive --cover-branches \
-		--ignore-files=search_indexes.py --ignore-files=highlight.py\
```
Comment (muhammad-ammar) [inline Makefile]: django-nose is having issue with `--ignore-files`. https://github.com/django-nose/django-nose/issues/240

File: notesapi/v1/tests/test_views.py [hunk 2]
```diff
@@ -92,10 +92,10 @@ def _get_search_results(self, **kwargs):
...
         Helper for search method. All keyword parameters are passed in GET
         """
-        q = QueryDict("user=" + TEST_USER, mutable=True)
-        q.update(kwargs)
-        url = reverse('api:v1:annotations_search') + '?{}'.format(q.urlencode())
-        result = self.client.get(url)
+        data = {"user": TEST_USER}
+        data.update(kwargs)
+        url = reverse('api:v1:annotations_search')
+        result = self.client.get(url, data=data)
```
Comment (muhammad-ammar) [inline notesapi/v1/tests/test_views.py]: Removed `QueryDict ` due to UnicodeEncodeErrors. `data` param to get method will be treated as [query_string](https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/test.py#L82])

**Comment by muhammad-ammar** [comment]
@symbolist @muzaffaryousaf PTAL. Thanks!

**Comment by muzaffaryousaf** [comment]
@muhammad-ammar Please mention the haystack issue here to keep track in future.

File: requirements/base.txt [hunk 3]
```diff
@@ -1,10 +1,10 @@
-Django==1.7.3
+Django>=1.8,<1.9
 requests==2.4.3
-djangorestframework==3.0.2
+djangorestframework==3.2.3
 django-rest-swagger==0.2.0
 django-haystack==2.3.1
 elasticsearch==0.4.5  # only 0.4 works for ES 0.90
-django-cors-headers==0.13
+django-cors-headers==1.1.0
```
Comment (muzaffaryousaf) [inline requirements/base.txt]: Why we have updated the package, were there failures/exceptions ?

**Comment by muzaffaryousaf** [comment]
@muhammad-ammar LGTM except the issues that you mentioned. I hope you tackled all the warnings that appear on running the server. :+1:

File: Makefile (hunk 1 above)
Comment (symbolist) [inline Makefile]: Is this not fixed in a newer version? If not, can we comment out the line and add an explanation instead?

**Comment by symbolist** [comment]
:+1: besides one comment.

File: Makefile (hunk 1 above)
Comment (muhammad-ammar) [inline Makefile]: @symbolist this is not fixed in latest version. I will comment the line and add an explanation.

File: requirements/base.txt (hunk 3 above)
Comment (muhammad-ammar) [inline requirements/base.txt]: @muzaffaryousaf Updated to avoid `RemovedInDjango19Warning`

File: notesserver/settings/common.py [hunk 4]
```diff
@@ -44,6 +44,8 @@
 )
 
 INSTALLED_APPS = [
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
```
Comment (muhammad-ammar) [inline notesserver/settings/common.py]: Added to remove `RemovedInDjango19Warning`

File: pylintrc [hunk 5]
```diff
@@ -315,10 +302,6 @@ max-public-methods=20
 
 [CLASSES]
 
-# List of interface methods to ignore, separated by a comma. This is used for
-# instance to not check methods defines in Zope's Interface base class.
-ignore-iface-methods=isImplementedBy,deferred,extends,names,namesAndDescriptions,queryDescriptionFor,getBases,getDescriptionFor,getDoc,getName,getTaggedValue,getTaggedValueTags,isEqualOrExtendedBy,setTaggedValue,isImplementedByInstancesOf,adaptWith,is_implemented_by
-
 # List of method names used to declare (i.e. assign) instance attributes.
```
Comment (muhammad-ammar) [inline pylintrc]: Removed because these are deprecated and obsolete with pylint==1.5.0

**Comment by muhammad-ammar** [comment]
@muzaffaryousaf @symbolist I have made more changes to remove warnings and fix the sudden build failure. Please have a final look.

I have created https://openedx.atlassian.net/browse/TNL-3851 to track the remaining issues.

File: requirements/base.txt [hunk 6]
```diff
@@ -1,10 +1,10 @@
-Django==1.7.3
+Django>=1.8,<1.9
```
Comment (symbolist) [inline requirements/base.txt]: This should be fixed to a specific version.

**Comment by muhammad-ammar** [comment]
@symbolist fixed the django version

**Comment by symbolist** [comment]
:+1:

**Pull request merged**